### PR TITLE
Fix orange statuses, run tests for [AppveyorCi AzureDevOps Bitbucket CircleCi Gitlab Readthedocs Shippable Wercker]

### PR DESCRIPTION
**Pull request by Thilas**
I guess because of the missing message, I just get a `vendor unresponsive` badge.

### CODE REVIEW

**Comment by Thilas** [comment]
Sure, sorry about that!

When generating a badge that should return a partially succeeded status (with Azure DevOps in my case), I get the following badge: 
image

Expected: 
image

**Comment by calebcartwright** [comment]
Thanks, can you also include the links (both for your badge and your Azure DevOps Build)?

**Comment by Thilas** [comment]
Sorry but my badge is on a private ( company) build.

**Comment by calebcartwright** [comment]
So are you running your own self-hosted Shields instance?

**Comment by calebcartwright** [comment]
That's what I thought 😄 

The main Shields.io service can only provide Test Result and Code Coverage badges for public Azure DevOps projects. If you want to have Test Result and Code Coverage badges from pipelines in **private** Azure DevOps then you'll have to run your own Shields instance. 

This is because Azure DevOps APIs for Test Results and Code Coverage require user/organization specific  authentication information. 

There was a similar discussion in https://github.com/badges/shields/issues/2502 that may be of interest for you

**Comment by Thilas** [comment]
No, that's not it. When running current code, I get the following exception:
```
[1] "message" is required
    at Object.exports.process (c:\Personal\Programmes\badges\shields\node_modules\joi\lib\errors.js:203:19)
    at internals.Object._validateWithOptions (c:\Personal\Programmes\badges\shields\node_modules\joi\lib\types\any\index.js:764:31)
    at module.exports.internals.Any.root.validate (c:\Personal\Programmes\badges\shields\node_modules\joi\lib\index.js:147:23)
    at module.exports.internals.Any.root.attempt (c:\Personal\Programmes\badges\shields\node_modules\joi\lib\index.js:177:29)
    at module.exports.internals.Any.root.assert (c:\Personal\Programmes\badges\shields\node_modules\joi\lib\index.js:172:14)
    at Function.invoke (c:\Personal\Programmes\badges\shields\services\base.js:319:11)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

But the build status received from Azure DevOps is correct (e.g. partially succeeded in my case).

**Comment by calebcartwright** [comment]
Yeah the message change still needs to go through 👍  The test results/coverage part is true but is related to another Azure DevOps issue I'm working on right now

**Comment by calebcartwright** [comment]
Would you also add a test (presumably a mocked one) to the azure-build.tester with the `partially succeeded` response? Looks like we were missing that before which would've caught this

**Comment by Thilas** [comment]
Sure

File: services/azure-devops/azure-devops-build.tester.js [hunk 1]
```diff
@@ -43,3 +43,14 @@ t.create('unknown project')
 t.create('unknown user')
   .get('/azure-devops/build/notarealuser/foo/515.json')
   .expectJSON({ name: 'build', value: 'user or project not found' })
+
+t.create('partially succeeded build')
+  .get(
+    '/azure-devops/build/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/4.json'
```
[the remark this answers is not in the document]
Comment (Thilas) [reply services/azure-devops/azure-devops-build.tester.js]: That's the case.

[the remark this answers is not in the document]
Comment (calebcartwright) [reply services/azure-devops/azure-devops-build.tester.js]: 👍

[the remark this answers is not in the document]
Comment (Thilas) [reply services/azure-devops/azure-devops-build.tester.js]: (FYI, this project is public, I created it specifically for Shields)

[the remark this answers is not in the document]
Comment (Thilas) [reply services/azure-devops/azure-devops-build.tester.js]: https://dev.azure.com/totodem/Shields.io/_build?definitionId=4

File: services/azure-devops/azure-devops-build.tester.js [hunk 2]
```diff
@@ -43,3 +43,14 @@ t.create('unknown project')
...
+  )
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'build',
+      value: isBuildStatus,
```
Comment (calebcartwright) [inline services/azure-devops/azure-devops-build.tester.js]: Just to fully cover the partially succeeded scenario explicitly, let's add `?style=_shields_test` param on the get, and then update this to use `expectJSON` and validate name, value, and colorB (`build`, `passing`, and `orange`)

File: services/azure-devops/azure-devops-build.tester.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (calebcartwright) [reply services/azure-devops/azure-devops-build.tester.js]: Yup, I was looking at it, but as an anonymous user I just can't retention

File: services/azure-devops/azure-devops-build.tester.js (hunk 2 above)
Comment (Thilas) [reply services/azure-devops/azure-devops-build.tester.js]: Done

**Comment by paulmelnikow** [comment]
Nice work on this!

**Pull request merged**